pkg/reconciler/knativeserving/knativeserving.go

```diff
@@ -124,9 +124,10 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ks *v1beta1.KnativeServi
 		r.appendExtensionManifests,
 		r.transform,
 		manifests.Install,
-		manifests.SetManifestPaths, // setting path right after applying manifests to populate paths
-		common.CheckDeployments,
+		manifests.SetManifestPaths,    // setting path right after applying manifests to populate paths
+		common.CheckWebhookDeployment, // Wait for webhook to be ready before creating Certificate resources
```

Comment:
knativeeventing.go also need this function, please add it there as well.
The name of the deployment resource for eventing is "eventing-webhook", you probably need to make the name as a param to the function "CheckWebhookDeployment".

Reply:
Are you sure about eventing needing this as well?

Serving needed it as there are two install steps where the first one blocked the second one (see line 126 and 129):
https://github.com/knative/operator/blob/00cda2a1d39ec2cbaa47bc4fe4471963ecb1d74c/pkg/reconciler/knativeserving/knativeserving.go#L126-L132

Eventing has only one install step which means nothing is broken (see line 143):
https://github.com/knative/operator/blob/00cda2a1d39ec2cbaa47bc4fe4471963ecb1d74c/pkg/reconciler/knativeeventing/knativeeventing.go#L143-L148

Reply:
Thanks for the explanation!